### PR TITLE
Correctly detect that an empty files page is loaded

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
1) Be more specific about which div the ``emptycontent`` id is contained in (there are lots of ``emptycontent`` divs in different parts of the page)
2) Move check for fileList being visible so it only applies when there are files.
3) Add a core files test creating an empty folder and navigating into it.
4) Add a core trashbin test that checks that an empty delete folder goes to the trashbin as an empty folder
5) For the "name-parts" code, correctly handle the case when there is only 1 part given (it was crashing in that case, we do not use that because it is a bit odd to specify file name parts and then only put 1 part, but may as well fix it)
6) In ``FileRow`` match the ``fileLinkXpath`` differently so that it works on TrashbinPage as well as FilesPage.

## Related Issue
#29138 
(this PR is a "downstream" thing that was noticed while getting Edge tests running)

## Motivation and Context
When a folder is empty, the UI test code is not correctly detecting that the (empty) files page for that folder is loaded. The "wait till page is loaded" logic was timing out (10 seconds). Now that we throw an exception on page load timeout, this was causing test fails in the ``files_texteditor`` tests, because those tests navigated into an empty folder and then created a file in the folder.

There were no core UI tests that navigated into an empty folder. We should have at least one such test so that the empty files page logic is exercised in core testing.

## How Has This Been Tested?
Run ``files_texteditor`` UI tests locally and confirm they now pass.'
Run the new core test locally and confirm it fails before the change, and passes with the changed code.
Travis will do a full test run for us.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by individual-it** [comment]
does this solve #29264 ?

**Comment by phil-davis** [comment]
Yes - I did not even know about that issue.
I am also adding some tests to navigate into a folder in the trashbin, which will exercise the empty file list in the trashbin page case. Another commit will come...

**Review by individual-it — APPROVED** [review]

**Comment by phil-davis** [comment]
@individual-it please check the diffs again. I added core tests and a lot of associated refactoring.

File: tests/ui/features/files/files.feature [hunk 1]
```diff
@@ -19,4 +19,13 @@ Feature: files
...
 		|'सिमप्ले फोल्देर $%#?&@'|
 		|'"somequotes1"'|
-		|"'somequotes2'"|
\ No newline at end of file
+		|"'somequotes2'"|
+
+	Scenario: Create a folder inside another folder
+		When I create a folder with the name "top-folder"
+		And I open the folder "top-folder"
+		Then there are no files/folders listed
```
Comment (phil-davis) [inline tests/ui/features/files/files.feature]: This step is a kind-of obvious test! But it does exercise the step code on a files-page file-folder list.
(the trashbin test added further down exercises the same step code when on a trashbin page)

**Review by individual-it — APPROVED** [review]

**Review by individual-it — CHANGES_REQUESTED** [review]
screenshot-2017-10-25 job 13786 2 - owncloud core - travis ci

**Comment by phil-davis** [comment]
@individual-it please review again. Then I can squash the last "fixup things that I broke" commit into the 2nd-last commit and let CI run again.

**Review by individual-it — APPROVED** [review]

**Comment by phil-davis** [comment]
Backport stable10 in #29353
